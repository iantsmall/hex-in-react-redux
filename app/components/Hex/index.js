/**
 *
 * Hex
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { corners } from 'utils/hexgrid';
// import styled from 'styled-components';

const Hex = props => {
  const { type, x, y, size } = props;
  const points = corners(type, x, y, size)
    .map(p => p.join(','))
    .join(' ');
  return <polygon {...props} points={points} />;
};

Hex.propTypes = {
  type: PropTypes.oneOf(['pointy-topped', 'flat-topped']).isRequired,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  size: PropTypes.number.isRequired,
};

export default memo(Hex);
