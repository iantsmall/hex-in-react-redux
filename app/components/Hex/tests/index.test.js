/**
 *
 * Tests for Hex
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
// import { render } from 'react-testing-library';
import render from 'utils/testRenderWithContext';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Hex from '../index';

const props = {
  type: 'pointy-topped',
  x: 1,
  y: 2,
  size: 255,
};

describe('<Hex />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Hex {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it.skip('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Hex {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
