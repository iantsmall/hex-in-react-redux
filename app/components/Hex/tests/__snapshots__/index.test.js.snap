// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hex /> Should render and match the snapshot 1`] = `
<polygon
  points="221.83647796503186,129.5 1.0000000000000155,257 -219.8364779650318,129.50000000000009 -219.8364779650319,-125.49999999999993 0.9999999999999531,-253 221.83647796503178,-125.50000000000011"
  size="255"
  type="pointy-topped"
  x="1"
  y="2"
/>
`;
